### PR TITLE
Add support for Apache Commons Lang ArrayUtils

### PR DESCRIPTION
**Pull request by pwntester**

### CODE REVIEW

**Comment by aschackmull** [comment]
Needs autoformat. Otherwise LGTM.

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 1]
```diff
@@ -1,27 +1,58 @@
 /** Definitions related to the Apache Commons Lang library. */
 
 import java
+private import semmle.code.java.dataflow.FlowSteps
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:4]: This needs bidirectional import to work, i.e. you need to add `private import java/ql/src/semmle/code/java/frameworks/apache/Lang.qll` in `FlowSteps.qll`

**Review by aschackmull — DISMISSED** [review]

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 2]
```diff
@@ -1,27 +1,65 @@
...
+    this.hasName([
+        "clone", "nullToEmpty", "remove", "removeAll", "removeElement", "removeElements", "reverse",
+        "shift", "shuffle", "subarray", "swap", "toArray", "toMap", "toObject", "toPrimitive",
+        "toString", "toStringArray"
+      ]) and
+    src = 0
+    or
+    this.hasName("add") and
+    this.getNumberOfParameters() = 2 and
+    src = [0, 1, 2]
```
Comment (joefarebrother) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:59]: The 2 is redundent here if only parameters 0 and 1 exist.
```suggestion
    this.hasName("add") and
    this.getNumberOfParameters() = 2 and
    src = [0, 1]
```

File: java/ql/src/semmle/code/java/frameworks/apache/Lang.qll [hunk 3]
```diff
@@ -1,27 +1,65 @@
...
+ * A taint preserving method on `org.apache.commons.lang.ArrayUtils` or `org.apache.commons.lang3.ArrayUtils`
+ */
+private class ApacheLangArrayUtilsTaintPreservingMethod extends TaintPreservingCallable {
+  ApacheLangArrayUtilsTaintPreservingMethod() {
+    this.getDeclaringType() instanceof TypeApacheArrayUtils
+  }
+
+  override predicate returnsTaintFrom(int src) {
+    this.hasName(["addAll", "addFirst"]) and
+    src = [0 .. getNumberOfParameters()]
```
Comment (intrigus-lgtm) [inline java/ql/src/semmle/code/java/frameworks/apache/Lang.qll:48]: Isn't this off by one?
I.e. shouldn't it be `[0 .. getNumberOfParameters()-1]`?

**Review by aschackmull — APPROVED** [review]

**Comment by aschackmull** [comment]
Hmm, this should probably have had a change note. Could you add one in a new PR? (Cf. a similar change note on https://github.com/github/codeql/pull/5126)

**Comment by pwntester** [comment]
Added here https://github.com/github/codeql/pull/5170

**Pull request merged**